src/com/ms/silverking/cloud/zookeeper/ZooKeeperExtended.java

```diff
@@ -42,1024 +43,1081 @@
+    DataCallbackOp op;
+
+    op = new DataCallbackOp();
+    addAsyncCallbackOp(op);
+    return op;
+  }
+
+  private StatCallbackOp newStatCallbackOp() {
+    StatCallbackOp op;
+
+    op = new StatCallbackOp();
+    addAsyncCallbackOp(op);
+    return op;
+  }
+
+  private VoidCallbackOp newVoidCallbackOp() {
+    VoidCallbackOp op;
+
+    op = new VoidCallbackOp();
+    addAsyncCallbackOp(op);
+    return op;
+  }
+
+  private StringCallbackOp newStringCallbackOp() {
+    StringCallbackOp op;
+
+    op = new StringCallbackOp();
+    addAsyncCallbackOp(op);
+    return op;
+  }
+
+  ///////////////
+  // operations
+
+  public void setEphemeralInteger(String path, int data) throws KeeperException {
+    setEphemeral(path, NumConversion.intToBytes(data));
+  }
+
+  public void setEphemeral(String path, byte[] data) throws KeeperException {
+    if (exists(path)) {
+      set(path, data);
+    } else {
+      create(path, data, CreateMode.EPHEMERAL);
     }
-    
-    public VoidCallbackOp deleteAsync(String path) {
-        VoidCallbackOp  op;
-        
-        op = newVoidCallbackOp();
-        delete(path, ANY_VERSION, this, op.getRequestUUID());
-        return op;
+  }
+
+  public Stat set(String path, byte[] data, int version) throws KeeperException {
+    try {
+      return setData(path, data, version);
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
+  }
+
+  /**
+   * set for any version
+   *
+   * @param path
+   * @param data
+   * @return
+   * @throws KeeperException
+   */
+  public Stat set(String path, byte[] data) throws KeeperException {
+    return set(path, data, -1);
+  }
+
+  public Stat setString(String path, String data) throws KeeperException {
+    return set(path, data.getBytes());
+  }
 
-    public void deleteRecursive(String path) throws KeeperException {
-        deleteChildrenRecursive(path);
-        delete(path);
+  public Stat setInteger(String path, int x) throws KeeperException {
+    return set(path, NumConversion.intToBytes(x));
+  }
+
+  private List<ACL> getAclForPath(String path) {
+    List<ACL> aclToUse = acl.getAclForPath(path);
+    if (aclToUse == null) {
+      aclToUse = acl.getDefaultAcl();
     }
-    
-    private void deleteChildrenRecursive(String path) throws KeeperException {
-        List<String>    children;
-        List<VoidCallbackOp>   ops;
-        
-        children = getChildren(path);
-        for (String child : children) {
-            deleteChildrenRecursive(path +"/"+ child);
-        }
-        ops = new ArrayList<>(children.size());
-        for (String child : children) {
-            ops.add(deleteAsync(path +"/"+ child));
-        }
-        waitForCompletion(ops);
+    return aclToUse;
+  }
+
+  public String create(String path, byte[] data, CreateMode createMode) throws KeeperException {
+    try {
+      return create(path, data, getAclForPath(path), createMode);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)